rust/src/lightning.rs

```diff
@@ -974,7 +975,7 @@ async fn handle_ldk_events(
                     Entry::Occupied(mut e) => {
                         let payment = e.get_mut();
                         payment.status = HTLCStatus::Succeeded;
-                        payment.timestamp = get_timestamp();
+                        payment.updated_timestamp = get_timestamp();
```

Comment:
🔧  Since it is easy to make a mistake and forget to update the `updated_timestamp` every time we update the payment status, perhaps it would be a good idea to make the fields private and a public method to mutate the state, which automatically updates the `updated_timestamp`.

Reply:
that is indeed the goal, hence the API proposed had this done internally :)
I will try to hide as much of the public fields away in the next stage.